.eslintrc.js

```diff
@@ -1,31 +1,243 @@
 module.exports = {
   env: {
     node: true,
+    es6: true,
+    'jest/globals': true,
   },
-  extends: 'eslint:recommended',
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'plugin:jest/recommended',
+    'prettier/@typescript-eslint', // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    'plugin:prettier/recommended',
+  ],
+  parser: '@typescript-eslint/parser',
   parserOptions: {
+    project: 'tsconfig.json',
     ecmaVersion: 2020,
     sourceType: 'module',
     impliedStrict: true,
     ecmaFeatures: {
       jsx: true,
     },
   },
+  plugins: ['@typescript-eslint', 'jest', 'jsdoc'],
   rules: {
-    'no-console': ['error', { allow: ['warn', 'error', 'log'] }],
-    indent: ['error', 2],
-    'linebreak-style': ['error', 'unix'],
-    quotes: ['error', 'single'],
-    semi: ['error', 'never'],
-    'comma-dangle': [
+    'no-console': 'warn',
```

Comment:
`no-console` with`tslint` will make lint fail, so we should keep the same like that.

Reply:
Will be fixed by the incoming commit